### DISCUSSION TITLE
[Question][Lecture5]

### DISCUSSION THREAD

**Discussion by 08kmc09** [prosyslab-classroom/cs424-program-reasoning#275] [Q&A]
Name: Mincheol Kwon

YOUR QUESTION

I'm trying to proof first-logic example, on 28th slide of lecture 5.

Show F: `forall x. p(x,x) -> exists x. forall y. p(x,y)` is invalid.

I tried to solve with dividing into two cases. 1) I |/= forall x. p(x,x) 2) I|= exists x. forall y. p(x,y)

But for each case, I can only prove that the contradiction occurs when |/= forall x. p(x,x) and I|= exists x. forall y. p(x,y), Just showing this case is sufficient?

**Comment by KihongHeo** [comment]
Hi. I don't fully understand your question. Can we talk after the class today?

**Comment by 08kmc09** [comment]
Sorry professor, I missed your comment on this issue. Can we talk after the Chuseok break?

**Comment by KihongHeo** [comment]
Sure.

**Comment by 08kmc09** [comment]
As far as I know, to show some formula is invalid, assume the some interpretation exists that satisfies the given formula.

Then, for the implication rule, to show `I |= forall x. p(x,x) -> exists x. forall y. p(x,y)`, I have to show the contradiction for both cases. 

I |/= forall x. p(x,x) or I |= exists x. forall y. p(x,y)

1) I |/= forall x. p(x,x)
In this branch, I have to show the contradiction for I |= exists x. forall y. p(x,y) and I |/= exists x. forall y. p(x,y) each.

2) I |= exists x. forall y. p(x,y)
Also in this branch, I have to show the contradiction for I |= exists x. forall y. p(x,y) and I |/= exists x. forall y. p(x,y).

It was possible to show the contradiction for these cases.
for branch 1) I |/= forall x. p(x,x) /\  I |= exists x. forall y. p(x,y)
for branch 2) I |= exists x. forall y. p(x,y) /\ I |/= exists x. forall y. p(x,y)

But for each case, I can't show the contradiction for these cases.
for branch 1) I |/= forall x. p(x,x) /\ I |/= exists x. forall y. p(x,y)
for branch 2) I |= exists x. forall y. p(x,y) /\ I |= exists x. forall y. p(x,y)

Did I missing something? If this question is not clarified, can I ask you at the end of the next lecture?

**Comment by KihongHeo** [comment]
Hi,
Some misunderstanding here.

1. To show a formula is "valid" (i.e., true for all interpretations)
- Assume the formula is "invalid 
- Proof by contradiction (i.e., assume there exists an $I$ that makes the formula false)

2. To show a formula is "invalid" (i.e., not always true for all interpretations)
- Provide a counterexample

I will be happy to take more questions also next week.

**Comment by 08kmc09** [comment]
Thank you! Now I can understand.
